tools/build_defs/pkg/archive.py

```diff
@@ -340,7 +340,14 @@ def add_tar(self,
       p.wait()
       intar = tarfile.open(fileobj=f, mode='r:')
     else:
-      intar = tarfile.open(name=tar, mode='r:' + compression)
+      if compression in ['gz', 'bz2']:
+        # prevent performance issues due to accidentally-introduced seeks
+        # during intar traversal by opening in "streaming" mode. gz, bz2
+        # are supported natively by python 2.7 and 3.x
+        inmode = 'r|' + compression
+      else:
+        inmode = 'r:' + compression
+      intar = tarfile.open(name=tar, mode=inmode)
```

Comment:
This change isn't strictly necessary to achieve the performance improvement, but it protects against perf regressions in that if someone were to add behavior that forces a re-seek in the file (e.g., if someone were to add back the `.name` removed later in this file), there's an error thrown when streaming. I'm not married to it; happy to remove if folks thing that's better/safer.